modules/core/include/opencv2/core/check.hpp

```diff
@@ -0,0 +1,129 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#ifndef OPENCV_CORE_CHECK_HPP
+#define OPENCV_CORE_CHECK_HPP
+
+#include <opencv2/core/base.hpp>
+
+namespace cv {
+
+//! @cond IGNORED
+namespace detail {
+
+/** Returns string of cv::Mat depth value: CV_8U -> "CV_8U" or "CV_InvalidDepth" */
+CV_EXPORTS const char* getMatDepth(int depth);
+
+/** Returns string of cv::Mat depth value: CV_8U -> "CV_8U" or NULL */
+CV_EXPORTS const char* getMatDepth_(int depth);
```

Comment:
do we really need 2 functions?

Probably, the function shall be renamed to "depthToString()".

I would retain only the first function, and let it return `<invalid depth>` instead of `CV_InvalidDepth`, becase the latter is not a valid macro or whatever

Reply:
Renamed. First one function is moved into `cv::` namespace. Second stays in `cv::detail::`.

Error string is changed to `<invalid depth>`.
BTW, I would like to avoid angle brackets in messages, because many OpenCV users just copy-paste such messages into GitHub/other markdown editors "as is".

Example: it is here "<invalid depth>" <== (really here, try to edit this comment)

> Quote example: it is here "<invalid depth>" <== (really here, try to edit this comment)

```
Code block example: it is here "<invalid depth>" <== (really here, try to edit this comment)
```

BTW, Adding extra "spaces" allows to hack markdown editors: "< invalid depth >"